cumulus_etl/etl/config.py

```diff
@@ -121,3 +125,38 @@ def as_json(self):
             "timestamp": self.timestamp,
             "hostname": self.hostname,
         }
+
+
+def latest_codebook_id_from_configs(output_root: store.Root) -> str | None:
+    try:
+        config_root = store.Root(output_root.joinpath("JobConfig"))
+        timestamp_dirs = sorted(config_root.ls(), reverse=True)
+        config_path = config_root.joinpath(f"{timestamp_dirs[0]}/job_config.json")
+        config = common.read_json(config_path)
+        return config.get("codebook_id")
+    except Exception:
+        return None
+
+
+def validate_output_folder(output_root: store.Root, codebook_id: str) -> None:
+    """
+    Confirm the user isn't trying to use different PHI folders for the same output folder.
+
+    If they did that, they would end up with all new anonymized IDs and could double their resource
+    counts, since nothing would match from the previous run.
+
+    It's safe to have multiple output folders all using the same PHI folder. But not the other way
+    around.
+    """
+    saved_codebook_id = latest_codebook_id_from_configs(output_root)
+    if not saved_codebook_id:
+        return
+
+    # And compare against the new PHI dir
+    if saved_codebook_id != codebook_id:
+        errors.fatal(
+            f"The output folder '{output_root.path}' is already connected "
+            "to a different PHI folder. "
+            "You must always use the same PHI folder for a given output folder.",
+            errors.WRONG_PHI_FOLDER,
```

Comment:
nit, ignorable: Putting myself in a user's shoes for a second, if I got this error message, I'm not sure I would 100% know what to do about it.

I don't know what I would do about it without modifying the data model - like, if you had the last path of the PHI, you could say 'it was used for PHI in location (y), though that may have changed if it was moved.' I'm not sure if saying something like 'look for a log with the string xxx' might also be helpful?

Reply:
Ah totally fair - we do actually already record the phi folder in the same config file. So I've added a bit of logic to pull that out and print it too. :+1: